### DISCUSSION TITLE
Page header gets removed after applying annotations.

### DISCUSSION THREAD

**Discussion by varun-rf** [pymupdf/PyMuPDF#3652] [Looking for help]
Hi all,

I need help with annotations. I'm trying to redact some texts using  `add_redact_annot`. It successfully redacts the area but also removes the header image that I had added in the previous step.

Simplified version of the code I used to add header and footer images.
```python
img_xref = 0
for page in doc:
   prect = page.rect
   if hdr_logo:
           myrect = fitz.Rect(0, 0, prect.width, h_height)
            page.set_mediabox(fitz.Rect(0, 0, prect.width, prect.height + h_height))
            img_xref = page.insert_image(myrect, stream=img, height=h_height, width=h_width, xref=img_xref)
   
   if footer_logo:
            cb = page.cropbox
            myrect = fitz.Rect(0, prect.height, prect.width, prect.height + f_height)
            page.set_mediabox(cb + fitz.Rect(0, -1*f_height, 0, 0))
            img_xref = page.insert_image(myrect, stream=img, height=f_height, width=f_width, xref=img_xref)

doc.save(pdfpath, incremental=True, deflate=True, encryption=fitz.PDF_ENCRYPT_KEEP)
```

Code for redaction
```python
doc = fitz.open(tempf.filename)
pg = doc[0]
redact_coordinates = [211.4375, 174.203125, 399.4375, 204.203125]
rgb_color = {red: 9, green: 47, blue: 86}
# convering RGB to 0-1 scale
redact_color = (rgb_color['red']/255, rgb_color['green']/255, rgb_color['blue']/255)
pg.add_redact_annot(fitz.Rect(redact_coordinates[0], redact_coordinates[1], redact_coordinates[2], redact_coordinates[3]), fill=redact_color)
pg.apply_redactions()

doc.save(tempf.filename, incremental=True, deflate=True, encryption=fitz.PDF_ENCRYPT_KEEP)
doc.close()
```

Header coordinates: Rect(0.0, 0.0, 612.0, 107.87050359712231)
Footer coordinates: Rect(0.0, 899.8704833984375, 612.0, 974.7194042617468)

Any help will be greatly appreciated.

**Comment by JorjMcKie** [comment]
Please provide a reproducing file!
It is quite probable this is a duplicate of a known error.

**Reply by varun-rf** [reply]
Hey, thanks for replying. I have attached a sample file. Let me know if it helps.

Source File: [sample_resume.pdf](https://github.com/user-attachments/files/16087571/sample_resume.pdf)

File after redaction: [sample_resume(Formatted).pdf](https://github.com/user-attachments/files/16087519/sample_resume.Formatted.pdf)

**Comment by JorjMcKie** [comment]
I didn't follow all the coordinate computations in depth, but:
If you apply redaction without parameters like you did, then into all images overlapping a redaction holes will be punched.
That may be the reason here.
Us option `images=pymupdf.PDF_REDACT_IMAGE_NONE` to keep images untouched

**Reply by varun-rf** [reply]
Thank you for your suggestion. I had already tried using the `images=pymupdf.PDF_REDACT_IMAGE_NONE` option before posting here, but the issue still persisted. Additionally, I'm not sure how the header image could overlap with the redaction area, as both have different coordinates.

**Reply by JorjMcKie** [reply]
Can you please provide me with a file
1. containing these images
2. your redactions added - **_but not applied_**

So that example file should show the header / footer images and a number of red, crossed-out rectangles at places where you added redactions.

**Reply by varun-rf** [reply]
I've encountered an issue with a specific header image—it creates an empty space at the top but doesn't display. This isn't related to redaction; I apologize for the confusion. Any idea why this particular image isn't showing up? Thanks for your help!

Header Image - https://drive.google.com/file/d/1uFezFAMNCYUdZIZQdNHbkL9sx2wcUhaK/view?usp=sharing

**Reply by JorjMcKie** [reply]
Ok, this is a moving target then: so we are **_not_** dealing with redaction problems.
When using `page.insert_image` you do a lot of things that I find hard to understand: why do you modify the page's `MediaBox` at all? This has all sorts of consequences that can easily cause things to become invisible?

What is the purpose of setting width and height in `insert_image` - why don't you start with using the default parameters?

**Comment by JorjMcKie** [comment]
> Just some side remarks:
>   * You can create PyMuPDF's geometry objects also by directly passing in a list / tuple with the right length. So in your example above you could create a `Rect` directly via `fitz.Rect(redact_coordinates)`. No need to name all 4 number separately as you did.
>   * In all functions with geometry objects as parameters, you can directly use the corresponding list / tuple with the correct length. So for instance you can add a redaction annotation **_without_** first creating a `Rect` like this: `add_redact_annot(redact_coordinates, ...)`. All methods understand this and internally do the right things.
>
> This will save you a lot of coding effort!

**Reply by varun-rf** [reply]
Thank you for sharing this. I'll keep this in mind.
